### PR TITLE
RSDK-3139 - add slam dependencies into cartographer

### PR DESCRIPTION
**Pull request by kim-mishra**
### Changes
Removed go.viam.com/slam dependencies and explicitly put any dependencies inside this repo.

### Manual Testing
Run cartographer on the main branch and saw a cone of points on app. Switched to the feature branch, ran `make build` and `make install`, restarted viam server, and saw same results on app.

### CODE REVIEW

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,119 @@
+// Package config implements functions to assist with attribute evaluation in the SLAM service.
+// Warning: The code in this package might be transient. Stability is not guaranteed.
```
Comment (nicksanford) [inline config/config.go:2]: This warning can be removed

**Comment by EshaMaharishi** [comment]
Removing myself from this review - also, feel free to leave me off PR's for the full modularization project, since Nick is leading the project :)

File: integration_test.go [hunk 2]
```diff
@@ -21,13 +21,13 @@ import (
...
-	slamConfig "go.viam.com/slam/config"
-	"go.viam.com/slam/dataprocess"
-	slamTesthelper "go.viam.com/slam/testhelper"
 	"go.viam.com/test"
 
 	viamcartographer "github.com/viamrobotics/viam-cartographer"
+	vcConfig "github.com/viamrobotics/viam-cartographer/config"
+	"github.com/viamrobotics/viam-cartographer/dataprocess"
 	"github.com/viamrobotics/viam-cartographer/internal/testhelper"
+	vcTesthelper "github.com/viamrobotics/viam-cartographer/testhelper"
```
Comment (nicksanford) [inline integration_test.go:30]: could we move `"github.com/viamrobotics/viam-cartographer/testhelper"` into `"github.com/viamrobotics/viam-cartographer/internal/testhelper"`

File: sensors/utils/utils.go [hunk 3]
```diff
@@ -0,0 +1,52 @@
+// Package utils contains helper functions for the sensor implementations.
+// Warning: The code in this package might be transient. Stability is not guaranteed.
```
Comment (nicksanford) [inline sensors/utils/utils.go:2]: can we please remove all the warnings that this package might be transient?

File: sensors/utils/utils.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+		return "", errors.New("index out of bounds")
+	}
+	return sensors[index], nil
+}
+
+// GetPNGImage first attempts to get a lazy PNG image. If the image is not a lazy PNG, the
+// function expects it to be an YCBCR image. If it is neither a lazy PNG nor an YCBCR, the
+// function errors out. The returned function is a release function that must be called once
+// the caller of GetPNGImage is done using the image.
+func GetPNGImage(ctx context.Context, cam camera.Camera) ([]byte, func(), error) {
```
Comment (nicksanford) [inline sensors/utils/utils.go:28]: I don't think this is used

**Review by nicksanford — CHANGES_REQUESTED** [review]

**Comment by nicksanford** [comment]
Could you please test cartographer manually to confirm that it still behaves as expected after this change & list the manual testing steps you performed in the PR description?

**Review by nicksanford — APPROVED** [review]

**Pull request merged**